internal/types/document.go

```diff
@@ -245,6 +254,11 @@ func (d *Document) Get(key string) (any, error) {
 		return value, nil
 	}
 
+	// we may have the key lay down inside the document
+	if strings.Contains(key, ".") {
+		return d.GetByPath(NewPathFromString(key))
```

Comment:
At the very least, documentation should be updated.

But given that we have method GetByPath that uses Get already, and now have Get that uses GetByPath – I think this is too complex.

Let's stick with Has, Get, etc working with single keys and HasByPath, GetByPath, etc working with paths.